<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitReportReporter -->
<testsuite hostname="Shweta-PC" failures="3" tests="30" name="dsAlgo_TestClasses.Stack_TestClass" time="215.377" errors="1" timestamp="2025-03-13T18:05:10 EDT" skipped="9">
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="callValidLogInData" time="4.867"/>
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="validLoginDataBase" time="5.299"/>
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="validLoginDataBase" time="7.647">
    <skipped/>
  </testcase> <!-- validLoginDataBase -->
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="OperationsInStack" time="5.799"/>
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="Implementation" time="6.022"/>
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="Stack_Applications" time="5.602"/>
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInOperationsInStackPage" time="6.006">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInOperationsInStackPage -->
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInOperationsInStackPage" time="6.134">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInOperationsInStackPage -->
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInOperationsInStackPage" time="6.363">
    <failure message="Test failed: Expected an error message when clicking the &#039;Run&#039; button without entering code, but no message was displayed." type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Test failed: Expected an error message when clicking the 'Run' button without entering code, but no message was displayed.
at org.testng.Assert.fail(Assert.java:111)
at dsAlgo_TestClasses.Stack_TestClass.checkErrorMessageWhenExecutingWithoutenteringCodeInOperationsInStackPage(Stack_TestClass.java:75)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </failure>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInOperationsInStackPage -->
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="OperationsInStacktryeditor" time="6.484"/>
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="OperationsInStacktryeditor" time="6.667"/>
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="OperationsInStacktryeditor" time="6.949"/>
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="navigatebacktoOperationsinstack" time="6.450"/>
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInImplementation" time="6.137">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInImplementation -->
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInImplementation" time="6.938">
    <failure message="Test failed: Expected an error message when clicking the &#039;Run&#039; button without entering code, but no message was displayed." type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Test failed: Expected an error message when clicking the 'Run' button without entering code, but no message was displayed.
at org.testng.Assert.fail(Assert.java:111)
at dsAlgo_TestClasses.Stack_TestClass.checkErrorMessageWhenExecutingWithoutenteringCodeInImplementation(Stack_TestClass.java:106)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </failure>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInImplementation -->
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInImplementation" time="7.009">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInImplementation -->
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="Implementationtryeditor" time="11.604"/>
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="Implementationtryeditor" time="7.334"/>
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="Implementationtryeditor" time="7.354"/>
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="navigatebacktoImplementation" time="6.113"/>
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInApplication" time="6.475">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInApplication -->
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInApplication" time="6.531">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInApplication -->
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInApplication" time="6.410">
    <failure message="Test failed: Expected an error message when clicking the &#039;Run&#039; button without entering code, but no message was displayed." type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Test failed: Expected an error message when clicking the 'Run' button without entering code, but no message was displayed.
at org.testng.Assert.fail(Assert.java:111)
at dsAlgo_TestClasses.Stack_TestClass.checkErrorMessageWhenExecutingWithoutenteringCodeInApplication(Stack_TestClass.java:138)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </failure>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInApplication -->
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="Applicationtryeditor" time="7.224"/>
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="Applicationtryeditor" time="7.644"/>
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="Applicationtryeditor" time="7.607"/>
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="navigatebacktoStackApplication" time="7.504"/>
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="navigateToOperationsInStackPracticeQuestions" time="10.945">
    <skipped/>
  </testcase> <!-- navigateToOperationsInStackPracticeQuestions -->
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="navigateToOperationsInStackPracticeQuestions" time="10.960">
    <error message="no such element: Unable to locate element: {&quot;method&quot;:&quot;xpath&quot;,&quot;selector&quot;:&quot;./div&quot;}
  (Session info: chrome=131.0.6778.204)
For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
Build info: version: &#039;4.29.0&#039;, revision: &#039;5fc1ec94cb&#039;
System info: os.name: &#039;Windows 11&#039;, os.arch: &#039;amd64&#039;, os.version: &#039;10.0&#039;, java.version: &#039;23.0.1&#039;
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [228fbc11c929550bb08299b68c729d36, findElement {value=./div, using=xpath}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 131.0.6778.204, chrome: {chromedriverVersion: 131.0.6778.264 (2d05e315153..., userDataDir: C:\Users\mahaj\AppData\Loca...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:59572}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: windows, proxy: Proxy(), se:cdp: ws://localhost:59572/devtoo..., se:cdpVersion: 131.0.6778.204, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 228fbc11c929550bb08299b68c729d36" type="org.openqa.selenium.NoSuchElementException">
      <![CDATA[org.openqa.selenium.NoSuchElementException: no such element: Unable to locate element: {"method":"xpath","selector":"./div"}
  (Session info: chrome=131.0.6778.204)
For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
Build info: version: '4.29.0', revision: '5fc1ec94cb'
System info: os.name: 'Windows 11', os.arch: 'amd64', os.version: '10.0', java.version: '23.0.1'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [228fbc11c929550bb08299b68c729d36, findElement {value=./div, using=xpath}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 131.0.6778.204, chrome: {chromedriverVersion: 131.0.6778.264 (2d05e315153..., userDataDir: C:\Users\mahaj\AppData\Loca...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:59572}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: windows, proxy: Proxy(), se:cdp: ws://localhost:59572/devtoo..., se:cdpVersion: 131.0.6778.204, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 228fbc11c929550bb08299b68c729d36
at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:501)
at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:485)
at org.openqa.selenium.remote.ErrorCodec.decode(ErrorCodec.java:167)
at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:138)
at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:50)
at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:215)
at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:216)
at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:174)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:545)
at org.openqa.selenium.remote.ElementLocation$ElementFinder$2.findElement(ElementLocation.java:165)
at org.openqa.selenium.remote.ElementLocation.findElement(ElementLocation.java:59)
at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:368)
at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:362)
at org.openqa.selenium.support.pagefactory.DefaultElementLocator.findElement(DefaultElementLocator.java:68)
at org.openqa.selenium.support.pagefactory.internal.LocatingElementHandler.invoke(LocatingElementHandler.java:38)
at jdk.proxy2/jdk.proxy2.$Proxy21.getText(Unknown Source)
at dsAlgo_PageFactory.Stack_PageFactory.childelements(Stack_PageFactory.java:177)
at dsAlgo_TestClasses.Stack_TestClass.navigateToOperationsInStackPracticeQuestions(Stack_TestClass.java:167)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </error>
  </testcase> <!-- navigateToOperationsInStackPracticeQuestions -->
  <testcase classname="dsAlgo_TestClasses.Stack_TestClass" name="navigateToOperationsInStackPracticeQuestions" time="11.299">
    <skipped/>
  </testcase> <!-- navigateToOperationsInStackPracticeQuestions -->
</testsuite> <!-- dsAlgo_TestClasses.Stack_TestClass -->
